pontis-ui/src/components/Navigation/NavPopover.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { Fragment } from "react";
+import classNames from "classnames";
+import { Popover, Transition } from "@headlessui/react";
+import { ChevronDownIcon } from "@heroicons/react/solid";
+
+interface PopoverContent {
+  name: string;
+  description: string;
+  href: string;
+  icon: (props: React.ComponentProps<"svg">) => JSX.Element;
+}
+
+interface CallsToAction {
+  name: string;
+  href: string;
+  icon: (props: React.ComponentProps<"svg">) => JSX.Element;
+}
+
+interface NavPopoverProps {
+  buttonName: string;
+  content: PopoverContent[];
+  callsToAction: CallsToAction[];
+}
+
+const NavPopover: React.FC<NavPopoverProps> = ({
+  buttonName,
+  content,
+  callsToAction,
+}) => (
+  <Popover className="relative">
+    {({ open }) => (
+      <>
+        <Popover.Button
+          className={classNames(
+            open ? "text-slate-900" : "text-slate-500",
+            "group inline-flex items-center rounded-md bg-slate-50 text-base font-medium hover:text-slate-900 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-4"
+          )}
+        >
+          <span>{buttonName}</span>
+          <ChevronDownIcon
+            className={classNames(
+              open ? "text-slate-600" : "text-slate-400",
+              "ml-2 h-5 w-5 group-hover:text-slate-500"
+            )}
+            aria-hidden="true"
+          />
+        </Popover.Button>
+
+        <Transition
+          as={Fragment}
+          enter="transition ease-out duration-200"
+          enterFrom="opacity-0 translate-y-1"
+          enterTo="opacity-100 translate-y-0"
+          leave="transition ease-in duration-150"
+          leaveFrom="opacity-100 translate-y-0"
+          leaveTo="opacity-0 translate-y-1"
+        >
```

Comment:
Not sure where all we use Transition and if the styles are similar but this is a lot of customization and might be useful to abstract it away into a component?

Reply:
I know we also use on command palette e.g.

Reply:
Unfortunately the styles overlap only partly and the component is `Transition.Child` vs `Transition`

Reply:
Found a good solution!